test/e2e/v1alpha1/e2e_vulnerability_scanning_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e_test
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+
+	buildv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+)
+
+var _ = Describe("Vulnerability Scanning", func() {
+	var testID string
+	var err error
+	var buildRun *buildv1alpha1.BuildRun
+	var cbs *clusterBuildStrategyPrototype
+	var insecure bool
+
+	AfterEach(func() {
+		testBuild.DeleteBR(buildRun.Name)
+	})
+
+	Context("Scanning for vulnerabilities in container images", func() {
+		var outputImage string
+		BeforeEach(func() {
+			testID = generateTestID("vuln")
+			outputImage = fmt.Sprintf("%s/%s:%s",
+				os.Getenv(EnvVarImageRepo),
+				testID,
+				"latest",
+			)
+
+			// Assume we use secure registry unless it is a cluster
+			// internal registry used as part of the test setup
+			insecure = strings.Contains(outputImage, "cluster.local")
+			cbs = NewClusterBuildStrategyPrototype().
+				Name("crane-pull-" + testID).
+				BuildStep(buildv1alpha1.BuildStep{
+					Container: corev1.Container{
+						Name:       "crane-pull",
+						Image:      "gcr.io/go-containerregistry/crane:latest",
+						WorkingDir: "$(params.shp-source-root)",
+						SecurityContext: &corev1.SecurityContext{
+							RunAsUser:  pointer.Int64(1000),
+							RunAsGroup: pointer.Int64(1000),
+						},
+						Env: []corev1.EnvVar{
+							{Name: "DOCKER_CONFIG", Value: "/tekton/home/.docker"},
+							{Name: "HOME", Value: "/tekton/home"},
+						},
+						Command: []string{"crane"},
+						Args: []string{
+							"pull",
+							"--format=tarball",
+							"python:3.4-alpine",
```

Comment:
We need to create a copy/mirror of this image in our GitHub Container Registry I would say to be independent of that image being removed at some point.

Reply:
Preferably use an existing mirrored one that is already unmaintained and therefore vulnerable, like ghcr.io/shipwright-io/shipwright-samples/node:12.